### PR TITLE
Replace auth config with user settings in user profile

### PR DESCRIPTION
**Pull request by npetridis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx [hunk 1]
```diff
@@ -21,12 +21,8 @@ function getWeb3WalletAddress(user: UserResource): string {
   return '';
 }
 
-function checkIfSectionIsOn(section: string) {
-  return section === 'on' || section === 'required';
-}
-
 export function ProfileCard(): JSX.Element {
-  const { authConfig } = useEnvironment();
+  const { userSettings: { attributes } } = useEnvironment();
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx:25]: 🙃  ditto

File: packages/clerk-js/src/ui/userProfile/security/Security.tsx [hunk 2]
```diff
@@ -8,16 +8,15 @@ import { PageHeading } from 'ui/userProfile/pageHeading';
 import { ActiveDevicesCard } from './DevicesAndActivity/ActiveDevicesCard';
 
 export function Security(): JSX.Element {
-  const { authConfig } = useEnvironment();
+  const { userSettings: { attributes } } = useEnvironment();
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/userProfile/security/Security.tsx:11]: 🙃   https://assets.pokemon.com/assets/cms2/img/pokedex/full/132.png

File: packages/clerk-js/src/ui/userProfile/security/ChangePassword.tsx [hunk 3]
```diff
@@ -32,8 +32,7 @@ export function ChangePassword(): JSX.Element {
     }
   };
 
-  const showRemovePassword =
-    user.passwordEnabled && authConfig.password !== 'required';
+  const showRemovePassword = user.passwordEnabled && attributes.password.required;
```
[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/userProfile/security/ChangePassword.tsx:35]: Also let's add a test for this case, I think we don't cover it in the unit tests

[the remark this answers is not in the document]
Comment (npetridis) [reply packages/clerk-js/src/ui/userProfile/security/ChangePassword.tsx]: 566723f18286dcd1244c8609f6b776c9c5b391df

File: packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx (hunk 1 above)
Comment (npetridis) [reply packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx]: 506f7afb0337050ff4e2d654fa237eb45b137b18

File: packages/clerk-js/src/ui/userProfile/security/Security.tsx (hunk 2 above)
Comment (npetridis) [reply packages/clerk-js/src/ui/userProfile/security/Security.tsx]: 506f7afb0337050ff4e2d654fa237eb45b137b18

**Review by SokratisVidros — APPROVED** [review]

File: packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx [hunk 4]
```diff
@@ -101,9 +98,9 @@ export function ProfileCard(): JSX.Element {
   );
 
   const showWebWallet = !!web3Wallet;
```
[the remark this answers is not in the document]
Comment (npetridis) [reply packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx:100]: `web3Wallet` is the address of the web3 wallet retrieved like this
```
  const user = useCoreUser();
  const web3Wallet = getWeb3WalletAddress(user);
```
If we want to replace it with new user_settings it means that when `attributes.web_wallet.enabled` is enabled we also have a web3 wallet address correct? @gkats @SokratisVidros

[the remark this answers is not in the document]
Comment (SokratisVidros) [reply packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx:100]: Correct

[the remark this answers is not in the document]
Comment (npetridis) [reply packages/clerk-js/src/ui/userProfile/account/profileCard/ProfileCard.tsx]: e8bc8bf

**Pull request merged**